src/main/java/com/aws/greengrass/componentmanager/builtins/ArtifactDownloaderFactory.java

```diff
@@ -108,22 +111,34 @@ public void checkDownloadPrerequisites(List<ComponentArtifact> artifacts, List<C
                 componentIds.stream().map(ComponentIdentifier::getName).collect(Collectors.toList());
         for (ComponentArtifact artifact : artifacts) {
             // TODO : Use dedicated component type
-            if (artifact.getArtifactUri().getScheme().equalsIgnoreCase(ArtifactDownloaderFactory.DOCKER_SCHEME)) {
-                if (!componentNames.contains(DOCKER_MANAGER_PLUGIN_SERVICE_NAME)) {
-                    throw new MissingRequiredComponentsException(DOCKER_PLUGIN_REQUIRED_ERROR_MSG);
-                }
-                try {
+            try {
+                validateArtifactUri(artifact);
+                if (artifact.getArtifactUri().getScheme().equalsIgnoreCase(ArtifactDownloaderFactory.DOCKER_SCHEME)) {
+                    if (!componentNames.contains(DOCKER_MANAGER_PLUGIN_SERVICE_NAME)) {
+                        throw new MissingRequiredComponentsException(DOCKER_PLUGIN_REQUIRED_ERROR_MSG);
+                    }
                     Image image = Image.fromArtifactUri(artifact);
                     if (image.getRegistry().isEcrRegistry() && image.getRegistry().isPrivateRegistry()
                             && !componentNames.contains(TOKEN_EXCHANGE_SERVICE_TOPICS)) {
                         throw new MissingRequiredComponentsException(TOKEN_EXCHANGE_SERVICE_REQUIRED_ERROR_MSG);
                     }
-                } catch (InvalidArtifactUriException e) {
-                    throw new PackageLoadingException(
-                            String.format("Failed to download due to bad artifact URI: %s", artifact.getArtifactUri()),
-                            e);
                 }
+            } catch (InvalidArtifactUriException e) {
+                throw new PackageLoadingException(String
+                        .format("Failed to download due to bad artifact URI: %s", artifact.getArtifactUri()), e);
             }
         }
     }
+
+    private void validateArtifactUri(ComponentArtifact artifact) throws InvalidArtifactUriException {
+        if (artifact.getArtifactUri() == null) {
```

Comment:
Add the component name/version to the error message

Reply:
With the current method signature that mapping is not available. ComponentArtifact does not store the componentId/name.

Reply:
Is it possible to add? I can foresee that that will be confusing without knowing what component has the issue